File: src/main/java/ru/ioleksiv/telegram/bot/api/annotations/filter/telegram/EncryptedPassportElementFilter.java
package ru.ioleksiv.telegram.bot.api.annotations.filter.telegram;

import ru.ioleksiv.telegram.bot.api.annotations.filter.primitive.NotNullFilter;
import ru.ioleksiv.telegram.bot.api.annotations.filter.primitive.StringFilter;
import ru.ioleksiv.telegram.bot.api.model.annotation.AnnotationState;
import ru.ioleksiv.telegram.bot.api.model.objects.passport.EncryptedPassportElement;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
@Documented
public @interface EncryptedPassportElementFilter {

    AnnotationState status() default AnnotationState.ON;

    EncryptedPassportElement.Type type() default EncryptedPassportElement.Type.ALL;

    NotNullFilter data() default @NotNullFilter(status = AnnotationState.OFF);

    StringFilter phoneNumber() default @StringFilter(status = AnnotationState.OFF);

    StringFilter email() default @StringFilter(status = AnnotationState.OFF);

    PassportFileArrayFilter fileArray() default @PassportFileArrayFilter(status = AnnotationState.OFF);

    PassportFileFilter frontSide() default @PassportFileFilter(status = AnnotationState.OFF);

    PassportFileFilter reverseSide() default @PassportFileFilter(status = AnnotationState.OFF);

    PassportFileFilter selfie() default @PassportFileFilter(status = AnnotationState.OFF);

}

File: core/src/main/kotlin/io/github/oleksivio/tl/kbot/core/annotations/filter/composite/EncryptedPassportElementFilter.kt
package io.github.oleksivio.tl.kbot.core.annotations.filter.composite

import io.github.oleksivio.tl.kbot.core.annotations.filter.primitive.NotNullFilter
import io.github.oleksivio.tl.kbot.core.annotations.filter.primitive.StringFilter
import io.github.oleksivio.tl.kbot.core.model.annotation.AnnotationState
import io.github.oleksivio.tl.kbot.server.api.objects.passport.EncryptedPassportElement

@Retention
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@MustBeDocumented
annotation class EncryptedPassportElementFilter(
    val status: AnnotationState = AnnotationState.ON,
    val type: EncryptedPassportElement.Type = EncryptedPassportElement.Type.UNKNOWN,
    val validator: Array<String> = [],
    val data: NotNullFilter = NotNullFilter(
        status = AnnotationState.OFF
    ),
    val phoneNumber: StringFilter = StringFilter(
        status = AnnotationState.OFF
    ),
    val email: StringFilter = StringFilter(
        status = AnnotationState.OFF
    ),
    val fileArray: PassportFileArrayFilter = PassportFileArrayFilter(
        status = AnnotationState.OFF
    ),
    val frontSide: PassportFileFilter = PassportFileFilter(
        status = AnnotationState.OFF
    ),
    val reverseSide: PassportFileFilter = PassportFileFilter(
        status = AnnotationState.OFF
    ),
    val selfie: PassportFileFilter = PassportFileFilter(
        status = AnnotationState.OFF
    )
)
